app/org/sagebionetworks/bridge/validators/UploadFieldDefinitionListValidator.java

```diff
@@ -0,0 +1,133 @@
+package org.sagebionetworks.bridge.validators;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.validation.Errors;
+
+import org.sagebionetworks.bridge.BridgeUtils;
+import org.sagebionetworks.bridge.models.upload.UploadFieldDefinition;
+import org.sagebionetworks.bridge.models.upload.UploadFieldType;
+import org.sagebionetworks.bridge.upload.UploadUtil;
+
+/**
+ * Validates a List of UploadFieldDefinitions. This doesn't use the Validator class, because (1) the typing for a List
+ * of UploadFieldDefinitions is a little wonky and (2) we needed to pass in a little more context to get this to work.
+ */
```

Comment:
This is fine. I usually pass it in via the constructor, which means we create the validator in a call and then dispose of it.

Reply:
I prefer the opposite: singletons with no state whose behavior can be determined entirely by their inputs.